src/com/hms/flexyosisoftconnector/OsisoftJsonPayload.java

```diff
@@ -346,7 +347,28 @@ public void completePayloadAttempt() {
             firstPayload = false;
           }
 
-          payload.append(PayloadBuilder.addContainerStartToOMFDataMessage(tagName));
+          String typeString = "";
+
+          if (tagPayload.getDataPoint().getType() == DataType.FLOAT) {
```

Comment:
Would it make more sense to implement tagPayload.getDataPoint().getOISTypeString()?  
You would need to extend DataPointBoolean, DataPointDword, .... and all the rest. It would only be one method. 

You would need to override a default that would return a string. Then you would only need to check for that string. A single If statement.

Reply:
I was talking to @TomKimsey about this and planned to put in in a constant in the json payload class for now. Adding it to the datapoint library would also make sense, maybe more sense than constants.

Reply:
in favor of keeping it to the osisoft connector floats do not translate to "float", instead they need to have the string "number" which is an OSIsoft OMF specific thing

Reply:
@oliver-walker-hms 
If you add a constant to the payload class, you still need to do if, else if, else if ..... too many ifs. 
The data type strings are unique to OSI. keep this in the OSI connector.  
Extend the DataPoint types to have a getTypeString() method.

Reply:
@it-hms the datapoint class is a generic non OSI class https://github.com/hms-networks/sc-flexy-datapoint-lib/blob/master/src/com/hms_networks/americas/sc/datapoint/DataPoint.java

Reply:
@TomKimsey 
The OSI connector is already dependent on this class. I don't think there's any reason the OSI connector can't make a child class that inherits from datapoint and adds OSI specific functionality.

Reply:
well it would be more readable in the library and probably useful in the future as well. The connector would just have a separate case for OSIsoft specific floats = numbers wherever the connector used it. I think having it in the library itself would be just fine as a getString version of getting the types

Reply:
@it-hms are you saying create 5 additional classes fjust to add in this special string handling? why not the much similar solution of having a method in OsisoftJsonPayload 
private string getOsiTypeString(Datatype dt)?

Reply:
Yes, actually. 5 classes, but each with one method. You could make the method way more purposeful. 
However, I would do this in a different refactoring PR. 
Eliminate the two calls to 
All this code would change to:
`
for (int i = startTagIndex; i < endTagIndex; i++) {
      if (TagInfoManager.getTagInfoList().get(i) != null) {
        TagInfo currentTag = (TagInfo) TagInfoManager.getTagInfoList().get(i);
        payload.append(currentTag.SpecialOSITagInfoMethod(getFlexyName(), GetFlexySerial())
`

Reply:
I like the idea of a generic datapoint.getTypeString() method added to the generic lib, but an extra class to add that functionality does not add enough value to warrant the existence of the class. 

Although not the perfect way but it could be reused for other applications would be 

**DataPoint.java**
`private string dataTypeString = "DEFAULT_VALUE_FOR_CLASS"
public static abstract string getTypeString(){};
public static abstract setTypeString(){};
`

Then the osisoft connector would just need to make one call to DataPointFloat.setTypeString() at the start to override the default.